tso-server/src/main/java/com/yahoo/omid/tso/PersistenceProcessor.java

```diff
@@ -23,6 +23,7 @@
     void persistAbort(long startTimestamp, boolean isRetry, Channel c, MonitoringContext monCtx);
 
     void persistTimestamp(long startTimestamp, Channel c, MonitoringContext monCtx);
-
-    void persistLowWatermark(long lowWatermark);
+    void persistLowWatermark(long lowWatermark, MonitoringContext monCtx);
+    void persistFlush(boolean forTesting);
```

Comment:
It's OK to have an interface method that behaves differently in real and in test env. It's not OK to have a param that control it as in this case. Please consider using a mock of PersistenceProcessor or partial mock of PersistenceProcessorImpl instead. It's a big read flag.